include/NeoFOAM/DSL/eqnSystem.hpp

```diff
@@ -0,0 +1,286 @@
+            explicitTerms_.push_back(eqnTerm);
+            break;
+        }
+    }
+
+    void addSystem(const EqnSystem<ValueType>& eqnSys)
+    {
+        for (auto& eqnTerm : eqnSys.temporalTerms_)
+        {
+            temporalTerms_.push_back(eqnTerm);
+        }
+        for (auto& eqnTerm : eqnSys.implicitTerms_)
+        {
+            implicitTerms_.push_back(eqnTerm);
+        }
+        for (auto& eqnTerm : eqnSys.explicitTerms_)
+        {
+            explicitTerms_.push_back(eqnTerm);
+        }
+    }
+
+    void solve()
+    {
+        bool allTermsEvaluated = evaluated();
+        if (!allTermsEvaluated)
+        {
+            if (fvSchemesDict.empty())
+            {
+                NF_ERROR_EXIT("No scheme dictionary provided.");
+            }
+            build(fvSchemesDict);
+        }
+
+        if (temporalTerms_.size() == 0 && implicitTerms_.size() == 0)
+        {
+            NF_ERROR_EXIT("No temporal or implicit terms to solve.");
+        }
+        if (temporalTerms_.size() > 0)
+        {
+            // integrate equations in time
+            fvcc::TimeIntegration<ValueType> timeIntergrator(
+                *this, fvSchemesDict.subDict("ddtSchemes")
+            );
+            timeIntergrator.solve();
+        }
+        else
+        {
+            // solve sparse matrix system
+        }
+    }
+
+    size_t size() const
+    {
+        return temporalTerms_.size() + implicitTerms_.size() + explicitTerms_.size();
+    }
+
+    // getters
+    const std::vector<EqnTerm<ValueType>>& temporalTerms() const { return temporalTerms_; }
+
+    const std::vector<EqnTerm<ValueType>>& implicitTerms() const { return implicitTerms_; }
+
+    const std::vector<EqnTerm<ValueType>>& explicitTerms() const { return explicitTerms_; }
+
+    std::vector<EqnTerm<ValueType>>& temporalTerms() { return temporalTerms_; }
+
+    std::vector<EqnTerm<ValueType>>& implicitTerms() { return implicitTerms_; }
+
+    std::vector<EqnTerm<ValueType>>& explicitTerms() { return explicitTerms_; }
+
+    const NeoFOAM::Executor& exec() const { return exec_; }
+
+    std::size_t nCells() const { return nCells_; }
+
+    fvcc::VolumeField<NeoFOAM::scalar>* volumeField()
+    {
+        if (temporalTerms_.size() == 0 && implicitTerms_.size() == 0)
+        {
+            NF_ERROR_EXIT("No temporal or implicit terms to solve.");
+        }
+        if (temporalTerms_.size() > 0)
+        {
+            volumeField_ = temporalTerms_[0].volumeField();
+        }
+        else
+        {
+            volumeField_ = implicitTerms_[0].volumeField();
+        }
+        return volumeField_;
+    }
+
+    NeoFOAM::scalar dt = 0;
+    NeoFOAM::Dictionary fvSchemesDict;
+
+private:
+
+    bool evaluated()
+    {
+        // check if terms have been evaluated
+        for (auto& eqnTerm : temporalTerms_)
+        {
+            if (!eqnTerm.evaluated())
+            {
+                return false;
+            }
+        }
+
```

Comment:
the loop could be refactored, and applied to all terms. Could look like this.
```suggestion
        // check if terms have been evaluated
        auto allEvaluated = [](auto &terms){
        for (auto& term : terms)
        {
            if (!eqnTerm.evaluated())
            {
                return false;
            }
        }
        return true;
        }
        if (!allEvaluated(implicitTerms_)
          || !allEvaluated(explicitTerms_)  
          || !allEvaluated(temporalTerms_))  {
          return false;
          }
       return true;
```